ndcube/asdf/converters/globalcoords_converter.py

```diff
@@ -0,0 +1,26 @@
+from typing import OrderedDict
+
+from asdf.extension import Converter
+
+
+class GlobalCoordsConverter(Converter):
+    tags = ["tag:sunpy.org:ndcube/global_coords/globalcoords-*"]
+    types = ["ndcube.global_coords.GlobalCoords"]
+
+    def from_yaml_tree(self, node, tag, ctx):
+        from ndcube.global_coords import GlobalCoords
+
+        globalcoords = GlobalCoords()
+        if node.get("internal_coords") is not None:
+            globalcoords._internal_coords = OrderedDict(node.get("internal_coords"))
+        globalcoords._ndcube = node["ndcube"]
+
+        return globalcoords
+
+    def to_yaml_tree(self, globalcoords, tag, ctx):
+        node = {}
+        node["ndcube"] = globalcoords._ndcube
+        if globalcoords._internal_coords:
+            node["internal_coords"] = dict(globalcoords._internal_coords)
```

Comment:
```suggestion
            node["internal_coords"] = globalcoords._internal_coords
```
`OrderedDict` is supported by asdf (via a yaml OMAP). Let me know if you ran into any issues trying to save a `OrderedDict`.

Reply:
Thanks for letting me know about the Yaml OMAP.